homeassistant/config.py

```diff
@@ -446,8 +446,11 @@ def _format_config_error(ex: vol.Invalid, domain: str, config: Dict) -> str:
             ]
             if has_api_password:
                 auth_conf.append({'type': 'legacy_api_password'})
-            if has_trusted_networks:
-                auth_conf.append({'type': 'trusted_networks'})
+            if trusted_networks:
+                auth_conf.append({
```

Comment:
I think that we might want to add a warning that users will need to migrate to a manual config.

Eventually all these things need to get away from the `http` component.

Reply:
I think warning is unnecessary for now, maybe an info log?

If user remove trusted_networks from http, it is a big breaking changes and will upset some users. We have to plan that removal more carefully.

Reply:
I agree. But we can start printing a warning that they should migrate?

Reply:
Add deprecated warning in http.trusted_networks config validation